src/main.c

```diff
@@ -167,10 +169,10 @@ static bool parse_args(int argc, char **args)
     prog_args = &args[optind];
     opt_prog_name = prog_args[0];
     if (opt_prof_data) {
-        char *prog_name = malloc(strlen(opt_prog_name) - 11);
-        strncpy(prog_name, opt_prog_name + 8, strlen(opt_prog_name) - 12);
-        prof_out_file = malloc(strlen(opt_prog_name) + 1);
-        sprintf(prof_out_file, "./prof/%s.prof", prog_name);
+        char *prog_basename = basename(opt_prog_name);
+        prof_out_file = malloc(strlen(prog_basename) + 11 + 1);
+        assert(prof_out_file);
+        sprintf(prof_out_file, "build/%s.prof", prog_basename);
```

Comment:
Is the leading `build/` necessary?

Reply:
It depends on working directory of `rv32emu` executable. There are 2 cases to profile:

1. 
```
// outside rv32emu directory
xxx/build/rv32emu -p xxx/build/xxx.elf
```

2. 
```
// inside build directory
./rv32emu -p xxx.elf
```

Without leading `build/`, the first case will always write to cwd which is parent directory of `xxx`. For second case, it is all right.

I think the new approach is concatenating the cwd and relative path (e.g., `xxx/build/` in first case and `./` in second case) to `rv32emu` executable. This concatenated path will always point to the `build` directory.

Reply:
> I think the new approach could be concatenating the cwd and relative path (e.g., `xxx/build/` in first case and `./` in second case) to `rv32emu` executable. This concatenated path will always point to the `build` directory.

I think the new one is better, and I expect a cleaner and elegant implementation.